### PR TITLE
[WIP] Costco store vault

### PR DESCRIPTION
**Pull request by unid15**
Added a huge store that can appear somewhere in space. It's operated by robots! 
In the store you can buy expensive things for cheap, and very rare things (switchtools, battering rams, tank transfer valves, ...) for lots of money. You can also buy circuitboards (75$ each) or clothing (150$ each). Stock is randomly generated every round

How to buy something:

It's guarded by lots of extremely robust and fast robots with 200 health, built-in rifle and 15-25 melee damage. Crossing the warning stripes while holding an unpaid item will [instantly lock down the entire store and send the guard bots after you](http://puu.sh/nNMxh/9a25a8f05a.jpg). 

If you don't have any money, you can get a drink at the bistro - alcohol and water is free there.

OTHER CHANGES:
- IDs on corpse landmarks now properly shows occupation
- existing_typesof now performs an additional check

TODO:
- Balance the loot and prices perhaps
- Look for bugs

### CODE REVIEW

**Comment by ihadtoregisterforthis** [comment]
## DEEPEST LORE

**Comment by Intigracy** [comment]
How about adding a table that continually respawns 1 nutriment "samples" after one is picked up.

Also, when the store goes into lockdown, does everyone inside get affected when it comes time for killy robots?

**Comment by Iamgoofball** [comment]
welcom to cosco

i lov u

File: vgstation13.dme [hunk 1]
```diff
@@ -1933,6 +1933,7 @@
 #include "maps\tgstation.dm"
 #include "maps\defficiency\areas.dm"
 #include "maps\defficiency\pipes.dm"
+#include "maps\randomvaults\costco.dm"
```
Comment (Intigracy) [inline vgstation13.dme]: bad dog

**Comment by unid15** [comment]
Everybody is killed, these robots don't take shit from anybody (I should probably change that, but it'd be difficult trying to determine the offender when, for example, an unpaid item is put in a crate and pushed out of the store)

**Comment by Intigracy** [comment]
Nah just have them kill everyone.

Does the store ever stop being in lockdown?

**Comment by unid15** [comment]
Nah

You're stuck in there forever unless you've got tools to deconstruct 2 rwalls

**Comment by Imagebard** [comment]
>Nah

Can't you just make the robots go back to their default positions and open everything back once everyone inside is dead?

**Comment by Imagebard** [comment]
Wait a minute
You're bolting people to the 1-tile big toilets too?

File: maps/randomvaults/costco.dm [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+			qdel(I)
+
+			src.loaded_cash += money_add
+			to_chat(user, "<span class='info'>You insert [money_add] space credits into \the [src]. \The [src] now holds [loaded_cash] space credits.</span>")
+	else
+		return ..()
+
+/mob/living/simple_animal/hostile/retaliate/costco_guardian
+	name = "Costco MERC-Bot"
+	desc = "Equipped with a ballistic weapon and a melee range shocker that is powerful enough to knock out a mega goliath through three layers of protection, this EMP-proof bot is not to be messed around with."
```
Comment (icantthinkofanameritenow) [inline maps/randomvaults/costco.dm:493]: That reminds me, is anyone still working on that mega-goliath?

**Comment by icantthinkofanameritenow** [comment]
If you can make a way to de-lockdown the place when the offender is dead, it should be good.

**Comment by ihadtoregisterforthis** [comment]
Is the stock randomized?
Why do those trees have snow while the grass doesn't?
Is it possible to have a small magazine with the prices?
Can I steal the bootyborgs?
What role does the baby hold?

Color a few tiles so it doesn't look so plain. Unless Costcos are that way.

**Comment by Shadowmech88** [comment]
Make sure to account for people using telescience/portals to steal things without taking them past the warning line.

File: maps/randomvaults/costco.dm [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+	if(directional_responses.Find(AM.dir))
+		say(directional_responses[AM.dir])
+
+/mob/living/simple_animal/robot/robot_cashier
+	name = "Cashier"
+	desc = "Only accepts cash."
+
+	icon = 'icons/mob/robots.dmi'
+	icon_state = "booty-red"
+
```
Comment (SkowronX) [inline maps/randomvaults/costco.dm:319]: You will regret this.

File: maps/randomvaults/costco.dm [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+	var/list/directional_responses = list(
+	NORTH = "Welcome to Costco. I love you.",
+	SOUTH = "Thank you for shopping at Costco, come again.",
+	)
+
+/mob/living/simple_animal/robot/robot_greeter/Die()
+	say("Vaae-#@ism is forbieeen at aostAo. Your aAAount will be terminated. Aonnection to aostAo mainframe lost... Self-destructing in 5... 4... 3...")
+	say("2... 1...")
+
+	spawn(3)
```
Comment (clusterfack) [inline maps/randomvaults/costco.dm]: spawn(3 SECONDS)

File: maps/randomvaults/costco.dm [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+	robogibs(get_turf(src))
+	qdel(src)
+
+/obj/effect/costco_entrance
+	name = "Costco entrance marker"
+
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "toddler"
+
+	invisibility = 0
```
Comment (clusterfack) [inline maps/randomvaults/costco.dm]: Use one of the defines for this

**Comment by unid15** [comment]
Stock is randomized (items are spawned from a pool). Also the shopping area is telescience-proof so don't worry about that

Updated the map. Robots now say "Welcome to Costco I love you", there's also a robot that can recite Costco's rules to you. Breaking a wall, window, structure or destroying an item now triggers the alarm and lockdown. Added a little maintenance area with cyborg rechargers, crematorium and a prison cell for a captive roboticist (who committed suicide, unfortunately for the robots). Also made the store more secure, with 3 layer walls - 1st layer is rwall with plasma glass covering, 2nd layer is rwall covered with reinforced girders and the 3rd layer is an ordinary wall. Will fix the trees later

File: maps/randomvaults/costco.dm (hunk 3 above)
Comment (Intigracy) [inline maps/randomvaults/costco.dm:319]: nah

**Comment by Intigracy** [comment]
Does preventing telescience from reaching it also prevent you using the hand tele?

If not, can you make it so you can't use the hand tele?

**Comment by Shadowmech88** [comment]
Is it 100% telescience proof? Even with a bluespace crystal in the telepad?

**Comment by unid15** [comment]
Shopping area is 100% telescienceproof, other areas require a crystal. Hand tele currently works but I'm going to fix it

Costco now features food samples! (thanks for the idea @Intigracy)
samples

**Comment by SkowronX** [comment]
:-1:  unless you replace borg sprites.

**Comment by Imagebard** [comment]
Skowron have you never heard of "Sex sells"?

**Comment by Intigracy** [comment]
There's nothing wrong with the borg sprite.

It's not players being shitters running around with it.

**Comment by gbasood** [comment]
Borg sprites are fine.
Even if shitters had the sprite it would be fine.
There are far worse things shitters ALREADY do every round than run around as a bootyborg.

**Comment by Imagebard** [comment]
>Even if shitters had the sprite it would be fine.

I'd politely disagree, but Iasking if this can be merged is more important right now.

**Comment by Imagebard** [comment]
Oh fuck its the !!CONFLICTS!!

**Comment by unid15** [comment]
there must be a way to add new vaults without causing conflicts with everybody else

**Comment by Intigracy** [comment]
There is if you put the defines on different lines than other people, I think

**Comment by PJB3005** [comment]
@unid15 could make vaults be loaded from external files at runtime but the issue is more complicated ones like this PR are gonna be fucked nontheless.

**Comment by Intigracy** [comment]
As you can see, my PR isn't conflicting.

#8951

Putting your defines at the end of the list like everyone else is your problem.

**Comment by Intigracy** [comment]
Also you still need to remove your map file from the DM's include.

**Comment by unid15** [comment]
Turned the outer walls into reinforced walls (because it's FAR too easy to steal items otherwise, even with 3 layered rwalls + rgirders). Added clothing (150$ each) and circuits (75$ each) sections to the store. All types of clothing and circuits can spawn there (except for the OP ones).

 

Gonna stop adding content now and focus on testing and bugfixing

**Comment by Intigracy** [comment]
This branch has conflicts that must be resolved

**Comment by unid15** [comment]
> [WIP]

**Comment by Intigracy** [comment]
:^)

**Comment by F-OS** [comment]
what is with the invulnerable walls, people should be able to break out some way.

**Comment by Intigracy** [comment]
None of the walls are invulnerable?

File: maps/randomvaults/costco.dmm [hunk 6]
```diff
@@ -1,176 +1,227 @@
 "aa" = (/turf/space,/area)
 "ab" = (/turf/simulated/wall,/area)
 "ac" = (/obj/structure/girder/reinforced,/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
-"ad" = (/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
-"ae" = (/obj/structure/window/reinforced/plasma,/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
-"af" = (/obj/structure/window/reinforced/plasma{dir = 4},/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
+"ad" = (/turf/simulated/wall/invulnerable,/area/vault/supermarket/entrance)
```
Comment (F-OS) [inline maps/randomvaults/costco.dmm:4]: that is the invulnerable wall

**Comment by Imagebard** [comment]
@Intigracy in the last image unid posted, the inner walls in the store are the same as the ones in CentComm.

**Comment by unid15** [comment]
@F-OS, it's far too easy to shoplift without invulnerable walls by standing next to an outer wall, putting walls/girders/any dense objects around you and slowly digging your way out

**Comment by PJB3005** [comment]
Also unid correct me if I'm wrong but if you do trigger lock down and survive you can still go through the airlock, right?

**Comment by Probe1** [comment]
Too literal, not SS13.  Rename it to something in-lore like Donk-Mart or whatever you particularly feel like.  There's actually a lot of well written lore hidden in an away mission file somewhere that Cael never finished coding.  

Anything in-universe except fuckin costco.

I like the idea.  I like the design - except the names.

**Comment by NigglyWiggly** [comment]
You can use one of my booty borg sprites if you don't wanna have twig-arms as a cashier

Also yeah change the name

**Comment by FudgePucker** [comment]
If the lockdown is too hard to implement and have it go down, maybe just have it gib shoplifters.

Also maybe put in a teleporter

**Comment by Shadowmech88** [comment]
W H E N
H
E
N

File: code/defines/obj/weapon.dm [hunk 7]
```diff
@@ -390,6 +390,10 @@
 	var/armed = 0
 	var/obj/item/weapon/grenade/iedcasing/IED = null
 
+/obj/item/weapon/legcuffs/beartrap/primed
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm:393]: /armed, for consistency.

**Comment by ihadtoregisterforthis** [comment]
Sometime someday

**Comment by unid15** [comment]
Hey that's funny, I was just thinking about reviving this as an away mission, instead of a vault

**Comment by Probe1** [comment]
that'd be neat.  I still like the idea as long as the naming convention fits SS13

**Comment by PJB3005** [comment]
This is basically dead, RIP.

**Comment by Intigracy** [comment]
RIP

**Comment by FudgePucker** [comment]
RIP
press f to pay respects.

**Pull request closed without merging**